### PR TITLE
Refactor/error handling

### PR DESCRIPTION
**Pull request by emorendorff**
# Description
Add checkForError to fetch and add conditional rendering to Poster component.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring

# How Has This Been Tested?
- [x] open localHost
- [ ] Cypress
- [x] dev tools

# Checklist:
- [ ] My code follows the Turing's guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] No console error messages

### CODE REVIEW

File: src/Components/Poster.js [hunk 1]
```diff
@@ -14,13 +14,13 @@ const Poster = ( {details, closePoster, formatRating, formatReleaseDate} ) => {
...
+      <p>Genre: {!details.genres ? "unavailable" : details.genres}</p>
+      <p>Avg Rating: {formatRating(rating)} </p>
       <p>Released {formatReleaseDate(releaseDate)}</p>
-      <p>{details.overview}</p>
+      <p>Overview: {!details.overview ? "No overview for this movie!" : details.overview}</p>
       <p>Runtime: {details.runtime} minutes </p>
-      <p>Budget: ${details.budget}</p>
-      <p>Revenue: ${details.revenue}</p>
+      <p>Budget: {!details.budget ? "unavailable" : details.budget}</p>
+      <p>Revenue: {!details.revenue ? "unavailable" : details.revenue}</p>
```
Comment (Meekb) [inline src/Components/Poster.js:23]: Great thinking on the conditional for what to display for missing data.

**Review by Meekb — APPROVED** [review]

File: src/Components/App.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ class App extends Component {
   }
   componentDidMount() {
     fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/`)
```
Comment (sertmer) [inline src/Components/App.js:19]: I'd take some time to review the rubric - fetch calls should be broken out into their own folder!

File: src/Components/Poster.js (hunk 1 above)
Comment (sertmer) [reply src/Components/Poster.js:23]: Good opportunity for destructuring to save you from typing details over and over - totally your call though!

**Review by sertmer — COMMENTED** [review]
Looks good so far! Thanks for keeping your changesets small!

**Pull request merged**